src/components/form/FormField.tsx

```diff
@@ -0,0 +1,46 @@
+import { FieldError, FieldValues, Path, UseFormRegister } from 'react-hook-form';
+
+import './FormField.scss';
+
+type FormFieldProps<T extends FieldValues> = {
+  label: string;
+  name: Path<T>;
+  type: string;
+  placeholder?: string;
+  register: UseFormRegister<T>;
+  error?: FieldError;
+  disabled?: boolean;
+  min?: string;
+};
+
+export const FormField = <T extends FieldValues>({
+  label,
+  name,
+  type,
+  placeholder,
+  register,
+  error,
+  disabled = false,
+  min,
+}: FormFieldProps<T>) => (
+  <>
+    <label htmlFor={name}>{label}</label>
+    <input
+      id={name}
+      placeholder={placeholder ?? ''}
+      type={type}
+      {...register(name)}
+      className="form-field"
+      disabled={disabled}
+      style={{
+        borderColor: error ? '#ff0000' : '#000',
+      }}
+      {...(type === 'date' && min ? { min } : {})}
```

Comment:
Should there be a different component for date types, or is `FormField` just a temporary component before it gets fleshed out further?

Reply:
I don't know, I guess right now it isn't necessary. If we do stick with the input type `date`, then we have an entire new component, where this single line is sufficient right now. Honestly, I've never used the input type `date` before, I've always relied on libs like [react-datepicker](https://www.npmjs.com/package/react-datepicker). I didn't want to add that though as you had already used the input `date` type. I think the decision might be made by any designer that comes in, also if there is a need for a datetime picker etc. Using the input `type` keeps the code small, the lib I mentioned can be large, but it is very widely used.